tests/test_qnode.py

```diff
@@ -1398,6 +1348,488 @@ def circuit(a):
             circuit.jacobian([1.0], method="A")
 
 
+class TestMetricTensor:
+    """Tests for metric tensor subcircuit construction and evaluation"""
+
+    def test_no_generator(self):
+        """Test exception is raised if subcircuit contains an
+        operation with no generator"""
+        dev = qml.device('default.qubit', wires=1)
+
+        def circuit(a):
+            qml.Rot(a, 0, 0, wires=0)
+            return qml.expval(qml.PauliX(0))
+
+        circuit = qml.QNode(circuit, dev)
+
+        with pytest.raises(QuantumFunctionError, match="has no defined generator"):
+            circuit.construct_metric_tensor([1])
+
+    def test_generator_no_expval(self, monkeypatch):
+        """Test exception is raised if subcircuit contains an
+        operation with generator that corresponds to no expectation value"""
```

Comment:
The term "expectation value" here confused me. Really this occurs if the generator cannot be cast to a known, evaluatable, observable op

Reply:
```suggestion
        operation with generator that corresponds to an object that is not an observable```